diff -au git-1.8.3.4/cache.h git-patch/cache.h
--- git-1.8.3.4/cache.h	2013-07-22 20:59:55.000000000 +0200
+++ git-patch/cache.h	2013-08-23 18:33:36.432853474 +0200
@@ -820,6 +820,8 @@
 extern void maybe_die_on_misspelt_object_name(const char *name, const char *prefix);
 extern int get_sha1_with_context(const char *str, unsigned flags, unsigned char *sha1, struct object_context *orc);
 
+extern int get_sha1_1(const char *name, int len, unsigned char *sha1, unsigned lookup_flags);
+
 typedef int each_abbrev_fn(const unsigned char *sha1, void *);
 extern int for_each_abbrev(const char *prefix, each_abbrev_fn, void *);
 
diff -au git-1.8.3.4/sha1_name.c git-patch/sha1_name.c
--- git-1.8.3.4/sha1_name.c	2013-07-22 20:59:55.000000000 +0200
+++ git-patch/sha1_name.c	2013-08-23 18:34:33.876849350 +0200
@@ -430,7 +430,7 @@
 	return 0;
 }
 
-static int get_sha1_1(const char *name, int len, unsigned char *sha1, unsigned lookup_flags);
+int get_sha1_1(const char *name, int len, unsigned char *sha1, unsigned lookup_flags);
 
 static int get_sha1_basic(const char *str, int len, unsigned char *sha1)
 {
@@ -735,7 +735,7 @@
 	return -1;
 }
 
-static int get_sha1_1(const char *name, int len, unsigned char *sha1, unsigned lookup_flags)
+int get_sha1_1(const char *name, int len, unsigned char *sha1, unsigned lookup_flags)
 {
 	int ret, has_suffix;
 	const char *cp;
